projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-list/query-list.component.ts

```diff
@@ -9,11 +10,81 @@ import { PartyMessagesGroup, QueryListData } from '../../domain';
 export class QueryListComponent implements OnChanges {
   @Input() public partyMessageGroup: PartyMessagesGroup;
   public queryListData: QueryListData | undefined;
+  public displayedColumns: column[] = [
+    { name: 'subject', displayName: 'Queries', sortOrder: SortOrder.UNSORTED },
+    { name: 'lastSubmittedBy', displayName: 'Last submitted by', sortOrder: SortOrder.UNSORTED },
+    { name: 'lastSubmittedDate', displayName: 'Last submission date', sortOrder: SortOrder.UNSORTED },
+    { name: 'lastResponseDate', displayName: 'Last response date', sortOrder: SortOrder.UNSORTED },
+    { name: 'lastResponseBy', displayName: 'Response by', sortOrder: SortOrder.UNSORTED }
+  ];
 
   public ngOnChanges(simpleChanges: SimpleChanges) {
     const currentPartyMessageGroup = simpleChanges.partyMessageGroup?.currentValue as PartyMessagesGroup;
     if (currentPartyMessageGroup) {
       this.queryListData = new QueryListData(currentPartyMessageGroup);
     }
   }
+
+  public sortTable(col: column) {
+    switch (col.displayName) {
+      case 'Queries': {
+        this.sort(col);
+        break;
+      }
+      case 'Last submitted by': {
+        this.sort(col);
+        break;
+      }
+      case 'Last submission date': {
+        this.sortDate(col);
+        break;
+      }
+      case 'Last response date': {
+        this.sortDate(col);
+        break;
+      }
+      case 'Response by': {
+        this.sort(col);
+        break;
+      }
+    }
```

Comment:
@DavidJayakumar - minor point, the linting might pick up that there's no `default` clause... might be worth adding one